<launch>
  <!-- <master auto="start"/> -->
  <param name="/use_sim_time" value="false"/>

  <arg name="image_debug" default="false" />
  <arg name="urdf" default="false" />

  <arg name="debug" default="false" />
  <arg unless="$(arg debug)" name="launch_prefix" value="" />
  <arg     if="$(arg debug)" name="launch_prefix" value="gdb --ex run --args" />

  <include file="$(find sbpl_perception)/config/roman_objects.xml"/>

  <param name="image_debug" value="false"/>

  <group if="$(arg urdf)">
      <include file="$(find roman_model)/robots/upload_rcta_roman.launch" />
  </group>

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find object_recognition_node)/rviz/realsense_camera_robot.rviz" required="true" />

  <rosparam command="load" file="$(find sbpl_perception)/config/pr2_env_config.yaml" />
  <rosparam command="load" file="$(find sbpl_perception)/config/pr2_planner_config.yaml" />
  <rosparam command="load" file="$(find sbpl_perception)/config/roman_camera_config.yaml" />

  <node pkg="object_recognition_node" type="object_localizer" name="object_localizer_service" output="screen" launch-prefix="mpirun -n 4 --mca mpi_yield_when_idle 1" respawn="false">
      <param name="image_debug" value="$(arg image_debug)"/>
      <param name="use_external_render" value="0"/>
      <param name="use_external_pose_list" value="0"/>
      <param name="use_icp" value="1"/>
      <param name="use_input_images" value="0"/>
  </node>

  <node pkg="object_recognition_node" type="obj_rec_node" name="object_recognition_node" output="screen" launch-prefix="$(arg launch_prefix)" respawn="false">
    <!-- <remap from="/keypress_topic" to="/obj_recognition_keyboard_listener/keydown"/> -->
    <!-- <remap from="input_cloud" to="/roman1/center_realsense/depth_registered/points"/> -->
    <remap from="input_cloud" to="/roman1/head_realsense/depth_registered/points"/>
    <param name="pcl_visualization" value="false"/>
    <param name="use_external_render" value="0"/>
    <param name="use_external_pose_list" value="0"/>
    <param name="use_icp" value="1"/>
    <param name="use_input_images" value="0"/>

    <!-- <param name="table_height" value="1.127"/> <!&#45;&#45;1.13&#45;&#45;> -->

    <param name="table_height" value="-0.083"/> <!--1.13-->

    <!--  Crate -->
    <param name="xmin" value="0.5"/> <!--1.13-->
    <param name="xmax" value="1.5"/> <!--1.13-->
    <param name="ymin" value="-0.05"/> <!--1.13-->
    <param name="ymax" value="0.50"/> <!--1.13-->

    <param name="reference_frame" value="/roman1/base_link"/>
    <param name="camera_optical_frame" value="/roman1/head_realsense_color_optical_frame"/>
    <param name="camera_frame" value="/roman1/head_realsense_depth_frame"/>
    <!-- <param name="camera_frame" value="/roman1/head_realsense_depth_frame"/> -->
  </node>

  <!-- <node pkg="keyboard" type="keyboard" name="obj_recognition_keyboard_listener" output="screen"> -->
  <!-- </node> -->

</launch>
